### PR TITLE
Expose pagination from headers 

### PR DESCRIPTION
**Pull request by pongstr**
ref #15 

- Fixed: Eslint Warnings comply with `no-var`, `prefer-const` rule…(as per .eslintrc)
- Added   : `parse-link-header` module to parse pagination info from `response.headers`
- Updated :  
     wraps the responses inside `{data: []}` object to be able to attach
     pagination information from `headers.link` to the response data.

### CODE REVIEW

**Comment by maxkoryukov** [comment]
@pongstr thank you for this PR

I have added several commits. Their main purpose: keep code simple, and reduce the amount of dependencies. And the most important thing: **backward compatibility**.

I think, we could create a **browser** compatible version of this SDK, so the final size of bundle is important (size depends on the amount of package-dependencies).

## What changed

* I have removed the new dependency `parse-link-header` 
* I have improved the CALLBACK calling routine. Now callback has 3 params:
  * err - error
  * data - raw Freshdesk response (parsed from JSON)
  * extra - some additional stuff, gathered from response.

The most useful thing for you - `extra` argument. It has two fields:
  * `pageIsLast` - boolean
  * `_headersLink` - string. But it is a workaround, I think, we should find more beatiful solution. I won't delete this field, to keep backward compatibility. But be aware: we still don't have `semver` rules for this repo (currently, **major** version matches the Freshdesk API version, and this is not convenient for maintenance of this package).

So, solutions for you: you could parse `extra._headersLink` in your callback-function.

**Comment by maxkoryukov** [comment]
And I don't like `data.data` members)))

**Comment by arjunkomath** [comment]
@maxkoryukov Is the docs updated for the `extra` parameter?

**Comment by maxkoryukov** [comment]
@arjunkomath **NO**. It is a shame... Sorry..

**Comment by maxkoryukov** [comment]
@arjunkomath doc updated see 1d4bd71

**Comment by maxkoryukov** [comment]
@pongstr , thanks for eslinting;)

**Comment by pongstr** [comment]
@maxkoryukov 

> I have added several commits. Their main purpose: keep code simple, and reduce the amount
> of dependencies. And the most important thing: backward compatibility.
>
> I think, we could create a browser compatible version of this SDK, so the final size of bundle 
> is important (size depends on the amount of package-dependencies).
>
> And I don't like data.data members)))

Understandable. 

> I won't delete this field, to keep backward compatibility. But be aware: we still don't have 
> semver rules for this repo (currently, major version matches the Freshdesk API version, and 
> this is not convenient for maintenance of this package).

Would be nice to have CONTRIBUTING guidelines (if you guys are open to that), I'm looking into 
contributing methods for [Solutions API](https://developers.freshdesk.com/api/#solutions), I'd like to take stub at it, maybe you can show me few pointers to get started?

**Comment by maxkoryukov** [comment]
@pongstr 

## contributing

Of course, open.. But need somebody, who will write `CONTRIBUTING.md` ;) 
Probably, some kind of `Code of conduct` will be useful tool...

## Implementing methods

As you have noticed [solutions](https://github.com/arjunkomath/node-freshdesk-api#solutions) are not implemented. But it is not a difficult task.

Here is an example of implementation: https://github.com/arjunkomath/node-freshdesk-api/blob/91e6dc026f96d369f6e79920f9e60965b5a0e10b/lib/client.js#L180

@arjunkomath had created a beautiful scaffold for the lib, so now it is very easy to implement the remaining stuff. My estimate: 30minutes for **Solutions** stuff (with documentation;) 

----

**Contributions are welcome!!!** :saxophone:

**Comment by pongstr** [comment]
@maxkoryukov I'd like to point out one thing though:

> `_headersLink` - string. But it is a workaround, I think, we should find more beatiful solution. 
> I won't delete this field, to keep backward compatibility. 

Does this mean that `extra._headersLink` should return the Header Link string? e.g.,
` <https://globalgaming.freshdesk.com/api/v2/tickets?per_page=10&page=2>; rel="next"`?

`_headersLink` is currently being assigned with `[extra.pageIsLast](https://github.com/arjunkomath/node-freshdesk-api/blob/master/lib/utils.js#L71)`

File: lib/utils.js [hunk 1]
```diff
@@ -54,20 +53,31 @@ function createResponseHandler (cb) {
...
 
-		let data = null;
+		let data = null
+		const extra = {
+			pageIsLast: true
+		}
 
 		debug('Got API response, status [%s]', response.statusCode)
 
+		if (response
```
Comment (pongstr) [inline lib/utils.js]:
```javascript
/**
  response and response.headers:
    - will always be present 
  response.headers.link:
    - will always be a string and is only set by, and is only set by the API if 
      the last page of object has been reached. see: 
      https://developers.freshdesk.com/api/#pagination
 */
  if (response.headers && response.headers.link) {
    extra.pageIsLast = false
    extra._headersLink = response.headers.link
  } 
```

Comment (maxkoryukov) [inline lib/utils.js]: my bad. fixed in 1f91095

**Comment by maxkoryukov** [comment]
my bad. fixed in 1f91095

**Comment by pongstr** [comment]
thank you! 👍

**Comment by maxkoryukov** [comment]
@pongstr thank **you** :laughing:

**Pull request merged**